app/src/main/java/com/fcaputo/parkingapp/mvp/view/SettingsView.kt

```diff
@@ -0,0 +1,76 @@
+package com.fcaputo.parkingapp.mvp.view
+
+import android.app.Activity
+import android.widget.TextView
+import androidx.appcompat.app.AppCompatActivity
+import com.fcaputo.parkingapp.R
+import com.fcaputo.parkingapp.databinding.ActivitySettingsBinding
+import com.fcaputo.parkingapp.mvp.contract.SettingsContract
+import com.fcaputo.parkingapp.mvp.view.base.ActivityView
+import com.fcaputo.parkingapp.utils.Constants
+import com.fcaputo.parkingapp.utils.validation.ValidationErrorType
+import com.google.android.material.dialog.MaterialAlertDialogBuilder
+
+class SettingsView(activity: Activity) : ActivityView(activity),SettingsContract.View {
+    private var binding: ActivitySettingsBinding = ActivitySettingsBinding.inflate(activity.layoutInflater)
+
+    init {
+        activity.setContentView(binding.root)
+    }
+
+    override fun onSaveButton(onClick: () -> Unit) {
+        binding.saveButton.setOnClickListener{
+            if(fieldsAreComplete()) {
+                onClick()
+            } else {
+                showErrorMessage(ValidationErrorType.INCOMPLETE_FIELD)
+            }
+        }
+    }
+
+    private fun fieldsAreComplete(): Boolean = listOf(binding.tInputParkingSize).all{ it.editText?.text?.isNotEmpty() ?: false }
+
+    override fun clear() { binding.tInputParkingSize.editText?.text?.clear() }
+
+    override fun getParkingSize(): Int = binding.tInputParkingSize.editText?.text.toString().toInt()
+
+    override fun showParkingSize(size: Int) {
+        val sizeString = activity?.resources?.getString(R.string.settings_current_size) + (if (size == Constants.SIZE_NOT_SET_INT) " -" else " $size")
+        binding.textCurrentSize.setText(sizeString, TextView.BufferType.NORMAL)
+    }
+
+    override fun showErrorMessage(error: ValidationErrorType) {
+        val (titleId, messageId) = getTitleMessageFromError(error)
+
+        MaterialAlertDialogBuilder(activity as AppCompatActivity)
+            .setTitle(titleId)
+            .setMessage(messageId)
+            .setPositiveButton(R.string.error_dialog_button_text, null)
+            .show()
+    }
+
+    private fun getTitleMessageFromError(error: ValidationErrorType): Pair<Int, Int> {
+        val titleId: Int
+        val contentId: Int
+
+        when(error){
```

Comment:
You can return directly this values to avoid to create titleId and contentId

Reply:
I decided to do this approach to avoid having multiple return sentences. I thought that one entry point and one return point was a cleaner way to write it. What do you think?

Reply:
You could:
```
return when (error) {
    ValidationErrorType.INCOMPLETE_FIELD -> {
        Pair(...,...)
    }
    ValidationErrorType.SIZE_IS_ZERO -> {
        Pair(...,...)
    }
}
```

Reply:
Ok, makes sense!

Reply:
Done